resize.js

```diff
@@ -86,13 +86,19 @@ image.checkCacheOrCreate = function (fileName, fileType, resolutionX, resolution
             return;
         }
         // It does not exist in the cache, so generate and upload
-        res.writeHead(200, {'Content-Type': supportedFileType(fileType)});
         image.encodeAndUpload(fileName, fileType, resolutionX, resolutionY, res);
     });
 };
 image.encodeAndUpload = function (fileName, fileType, resolutionX, resolutionY, res) {
     var file = config.get('originals_dir') + '/' + fileName;
+    if ( !fs.existsSync(file)) {
```

Comment:
You shouldn't use any `sync` calls ever (apart from possibly on startup). NodeJS is single-threaded, so any `sync` call will block the entire application (other than that it completely destroys performance, it also makes it trivial to DOS the server). Instead, use `fs.exists(cb)`.